bigquery/storage/managedwriter/options.go

```diff
@@ -48,31 +61,55 @@ type writerClientOption interface {
 	ApplyWriterOpt(*writerClientConfig)
 }
 
-// enableMultiplex enables multiplex behavior in the client.
-// maxSize indicates the maximum number of shared multiplex connections
-// in a given location/region
+// EnableMultiplexing is an EXPERIMENTAL option that enables connection sharing
+// when instantiating the Client.  Only writes to default streams can leverage the
+// multiplex pool.  Internally, the client maintains a pool of connections per BigQuery
+// destination region, and will grow the pool to it's maximum allowed size if there's
+// sufficient traffic on the shared connection(s).
 //
-// TODO: export this as part of the multiplex feature launch.
-func enableMultiplex(enable bool, maxSize int) option.ClientOption {
-	return &enableMultiplexSetting{useMultiplex: enable, maxSize: maxSize}
+// This ClientOption is EXPERIMENTAL and subject to change.
+func EnableMultiplexing(enable bool) option.ClientOption {
+	return &enableMultiplexSetting{useMultiplex: enable}
 }
 
 type enableMultiplexSetting struct {
 	internaloption.EmbeddableAdapter
 	useMultiplex bool
-	maxSize      int
 }
 
 func (s *enableMultiplexSetting) ApplyWriterOpt(c *writerClientConfig) {
 	c.useMultiplex = s.useMultiplex
+}
+
+// MaxMultiplexPoolSize is an EXPERIMENTAL option that sets the maximum
+// shared multiplex pool size when instantiating the Client.  If multiplexing
+// is not enabled, this setting is ignored.  By default, the limit is a single
+// shared connection.
+//
+// This ClientOption is EXPERIMENTAL and subject to change.
+func MaxMultiplexPoolSize(maxSize int) option.ClientOption {
+	return &maxMultiplexPoolSizeSetting{maxSize: maxSize}
+}
+
+type maxMultiplexPoolSizeSetting struct {
+	internaloption.EmbeddableAdapter
+	maxSize int
+}
+
+func (s *maxMultiplexPoolSizeSetting) ApplyWriterOpt(c *writerClientConfig) {
 	c.maxMultiplexPoolSize = s.maxSize
 }
 
-// defaultMaxInflightRequests sets the default flow controller limit for requests for
-// all AppendRows connections created by this client.
+// DefaultMaxInflightRequests is an EXPERIMENTAL ClientOption for controlling
+// the default limit of how many individual AppendRows write requests can
+// be in flight on a connection at a time.  This limit is enforced on all connections
+// created by the instantiated Client.
 //
-// TODO: export this as part of the multiplex feature launch.
-func defaultMaxInflightRequests(n int) option.ClientOption {
+// Note: the WithMaxInflightRequests WriterOption can still be used to control
+// the behavior for individual ManagedStream writers when not using multiplexing.
+//
+// This ClientOption is EXPERIMENTAL and subject to change.
+func DefaultMaxInflightRequests(n int) option.ClientOption {
```

Comment:
nit: most of our options are prefixed with With, could we carry that convention here as well?